### PR TITLE
feat: add Flipt to playground

### PR DESCRIPTION
**Pull request by lukas-reining**
## This PR

Adds Flipt to the playground as discussed with @markphelps.

@markphelps for being able to show "live changing" of flags I chose not to use the read-only mode with static flag definitions.
I chose to use SQLite and created a copy of a DB with configured flags that can be copied before doing `docker compose up`.
Otherwise Flipt works great with the Playground and is a great addition for showing live changes in an UI.

### CODE REVIEW

**Comment by markphelps** [comment]
cheers @lukas-reining !! thank you very much for adding Flipt to the playground!

I'm having trouble finding Flipt in the list of providers in the playground to switch between, am I doing something wrong?

CleanShot 2024-05-18 at 08 57 48

Note: I can access the Flipt UI and see that the flags / sqlite data has been imported correctly

CleanShot 2024-05-18 at 08 59 07

**Comment by lukas-reining** [comment]
> cheers @lukas-reining !! thank you very much for adding Flipt to the playground!
> 
> I'm having trouble finding Flipt in the list of providers in the playground to switch between, am I doing something wrong?
> 
> CleanShot 2024-05-18 at 08 57 48
> 
> Note: I can access the Flipt UI and see that the flags / sqlite data has been imported correctly
> 
> CleanShot 2024-05-18 at 08 59 07

Hey @markphelps, I guess you will need to comment out the image in the `docker-compose.yaml` for the demo and the fib service:

```yaml
services:
  demo:
    image: ghcr.io/open-feature/playground-app:v0.15.2 # x-release-please-version
    # build:
    #  dockerfile: ./packages/app/Dockerfile
    #  context: .
```

to 

```yaml
services:
  demo:
    # image: ghcr.io/open-feature/playground-app:v0.15.2 # x-release-please-version
    build:
    dockerfile: ./packages/app/Dockerfile
    context: .
```

Does it work now? :)

**Comment by markphelps** [comment]
> > cheers @lukas-reining !! thank you very much for adding Flipt to the playground!
> > I'm having trouble finding Flipt in the list of providers in the playground to switch between, am I doing something wrong?
> > CleanShot 2024-05-18 at 08 57 48
> > Note: I can access the Flipt UI and see that the flags / sqlite data has been imported correctly
> > CleanShot 2024-05-18 at 08 59 07
> 
> Hey @markphelps, I guess you will need to comment out the image in the `docker-compose.yaml` for the demo and the fib service:
> 
> ```yaml
> services:
>   demo:
>     image: ghcr.io/open-feature/playground-app:v0.15.2 # x-release-please-version
>     # build:
>     #  dockerfile: ./packages/app/Dockerfile
>     #  context: .
> ```
> 
> to
> 
> ```yaml
> services:
>   demo:
>     # image: ghcr.io/open-feature/playground-app:v0.15.2 # x-release-please-version
>     build:
>     dockerfile: ./packages/app/Dockerfile
>     context: .
> ```
> 
> Does it work now? :)

silly me, yes works great now thank you!

one idea instead of requiring users to copy over the sqlite db before running `docker-compose up` would be that we could take advantage of our `import` functionality: https://docs.flipt.io/operations/import-export#import to import the flag state from yaml.

we do this in our examples via an `init` service in docker-compose, ex: https://github.com/flipt-io/flipt/blob/main/examples/openfeature/docker-compose.yml#L11-L20

it might be a bit more portable this way, but im happy either way 😄

**Review by markphelps — APPROVED** [review]
thank you @lukas-reining !! looks great

**Comment by lukas-reining** [comment]
> > > cheers @lukas-reining !! thank you very much for adding Flipt to the playground!
> > > I'm having trouble finding Flipt in the list of providers in the playground to switch between, am I doing something wrong?
> > > CleanShot 2024-05-18 at 08 57 48
> > > Note: I can access the Flipt UI and see that the flags / sqlite data has been imported correctly
> > > CleanShot 2024-05-18 at 08 59 07
> > 
> > 
> > Hey @markphelps, I guess you will need to comment out the image in the `docker-compose.yaml` for the demo and the fib service:
> > ```yaml
> > services:
> >   demo:
> >     image: ghcr.io/open-feature/playground-app:v0.15.2 # x-release-please-version
> >     # build:
> >     #  dockerfile: ./packages/app/Dockerfile
> >     #  context: .
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > to
> > ```yaml
> > services:
> >   demo:
> >     # image: ghcr.io/open-feature/playground-app:v0.15.2 # x-release-please-version
> >     build:
> >     dockerfile: ./packages/app/Dockerfile
> >     context: .
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Does it work now? :)
> 
> silly me, yes works great now thank you!
> 
> one idea instead of requiring users to copy over the sqlite db before running `docker-compose up` would be that we could take advantage of our `import` functionality: https://docs.flipt.io/operations/import-export#import to import the flag state from yaml.
> 
> we do this in our examples via an `init` service in docker-compose, ex: https://github.com/flipt-io/flipt/blob/main/examples/openfeature/docker-compose.yml#L11-L20
> 
> it might be a bit more portable this way, but im happy either way 😄

Oh I have completely overseen the import function.
I switched to using it, basically exactly as you did in the linked code.
This is great and is even better for starters, thanks @markphelps.
What do you think @markphelps?

**Review by markphelps — APPROVED** [review]
double approve

**Comment by beeme1mr** [comment]
I'm not able to get it working locally. Flipt errors getting the db driver for dqlite. Here's what I'm seeing the logs.

```
playground-flipt-1            |     _________       __ 
playground-flipt-1            |    / ____/ (_)___  / /_
playground-flipt-1            |   / /_  / / / __ \/ __/
playground-flipt-1            |  / __/ / / / /_/ / /_  
playground-flipt-1            | /_/   /_/_/ .___/\__/  
playground-flipt-1            |          /_/           
playground-flipt-1            | 
playground-flipt-1            | Version: v1.42.0
playground-flipt-1            | Commit: 5ac6a4f16e8ac9d0eb931486b06fbddc62857d0d
playground-flipt-1            | Build Date: 2024-05-15T13:58:18Z
playground-flipt-1            | Go Version: go1.22.2
playground-flipt-1            | OS/Arch: linux/amd64
playground-flipt-1            | 
playground-flipt-1            | A newer version of Flipt exists at https://github.com/flipt-io/flipt/releases/tag/v1.42.1, 
playground-flipt-1            | please consider updating to the latest version.
playground-flipt-1            | Error: getting db driver for: sqlite3: unable to open database file: no such file or directory
playground-flipt-1 exited with code 1
```

The init script is also trying to mount `flipt.yml` but that doesn't exist at the root of the project.

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,5 @@ Thumbs.db
 # yalc stuff
 .yalc
 yalc.lock
+
+config/flipt/flipt.db
```
Comment (markphelps) [inline .gitignore:49]: dont need this anymore I think

File: docker-compose.yaml [hunk 2]
```diff
@@ -129,5 +132,30 @@ services:
...
       - "1031:1031"
 
+  init_flipt:
+    image: docker.flipt.io/flipt/flipt:v1.42.0
+    command: ["./flipt", "import", "/var/opt/flipt/flipt.yml"]
+    environment:
+      FLIPT_LOG_LEVEL: debug
+      FLIPT_META_TELEMETRY_ENABLED: false
+    volumes:
+      - "./flipt.yml:/flipt.yml"
```
Comment (markphelps) [inline docker-compose.yaml:142]: dont think we need this since it will be at `./config/flipt/flipt.yml` already

**Review by markphelps — COMMENTED** [review]
couple things related to @beeme1mr 's comments

**Comment by lukas-reining** [comment]
Yes, these were wrong, I completely oversaw those after changing to the import feature @markphelps.
Can you try again @beeme1mr?

**Comment by beeme1mr** [comment]
It runs now but the UI wasn't bootstrapped with the flags.

image

I'm looking into it now.

**Comment by beeme1mr** [comment]
The error message is:

```
playground-init_flipt-1       | 2024-05-21T19:09:25Z    DEBUG   configuration source    {"path": "/etc/flipt/config/default.yml"}
playground-init_flipt-1       | Error: getting db driver for: sqlite3: unable to open database file: no such file or directory
```

It looks like we're missing `default.yml` in the config folder.

**Comment by beeme1mr** [comment]
It's working for me now. Please try my changes to make sure it works for others.

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**